hw2/03-count.js

```diff
@@ -1,3 +1,33 @@
-// Add your code here
+document.addEventListener("DOMContentLoaded", () => {
+  const userInput = document.getElementById("userInput");
+  const textContainer = document.getElementById("textToHighlight");
+  const originalText = textContainer.textContent;
 
-input.addEventListener('keydown', handleKeyDown);
+  // Function to escape regex special characters to avoid unexpected behavior
+  const escapeRegex = (string) => {
+    return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+  };
+
+  // Function to highlight the word in the text container when enter is pressed
+  const highlightWord = (event) => {
+    if (event.key === "Enter") {
```

Comment:
I thought the program wasn't working until I read this.
As a user, I had no idea I had to press 'Enter'.
Maybe adding some text or a button to indicate this would be a good idea.

Though, then the html wouldn't match the reference screenshot, and I don't know if that's okay.

Or maybe changing the functionality to occur more frequently, like every 'keyup', would be better.

Reply:
Doesn't the **resource section** in the requirement specified to make use of "Keydown Event".?

That is why I used **Keydown** Event instead of **keyup**.